examples/tensorrt_llm/common/base_engine.py

```diff
@@ -368,3 +406,102 @@ async def async_llm_wrapper():
+        prefill_request.sampling_params["max_tokens"] = 1
+        prefill_request.disaggregated_params = DisaggregatedParams(
+            request_type="context_only"
+        )
+
+        if self._prefill_client is None:
+            raise ValueError("Prefill client not initialized")
+
+        # TODO: Use smart KV router to determine which prefill worker to use.
+        ctx_responses = [
+            ctx_response
+            async for ctx_response in await self._prefill_client.round_robin(
+                prefill_request.model_dump_json()
+            )
+        ]
+        if len(ctx_responses) > 1:
+            raise ValueError(
+                "Prefill worker returned more than one response. This is currently not supported in remote prefill mode."
+            )
+        logger.debug(
+            f"Received response from prefill worker: {ctx_responses[0].data()}"
+        )
+        ctx_response_obj = TRTLLMWorkerResponse.model_validate_json(
+            ctx_responses[0].data()
+        )
+        ctx_response_obj.outputs = [
+            TRTLLMWorkerResponseOutput(**ctx_response_obj.outputs[0])
+        ]
+        assert ctx_response_obj.outputs[0].disaggregated_params is not None
+
+        return ctx_response_obj
+
+    async def generate(self, request: TRTLLMWorkerRequest):
+        if self._llm_engine is None:
+            raise RuntimeError("Engine not initialized")
+
+        if self._error_queue and self._error_queue.qsize() > 0:
+            error = self._error_queue.get()
+            raise error
+
+        self._ongoing_request_count += 1
+
+        try:
+            worker_inputs = request.tokens.tokens
+
+            disaggregated_params = (
+                DisaggregatedTypeConverter.to_llm_disaggregated_params(
+                    request.disaggregated_params
+                )
+            )
+
+            if self._remote_prefill and self._server_type == ServerType.GEN:
+                ctx_response_obj = await self._get_remote_prefill_response(request)
+
+                worker_inputs = ctx_response_obj.prompt_token_ids
+                disaggregated_params = (
+                    DisaggregatedTypeConverter.to_llm_disaggregated_params(
+                        DisaggregatedParams(
+                            **ctx_response_obj.outputs[0].disaggregated_params
+                        )
+                    )
+                )
+                disaggregated_params.request_type = "generation_only"
+
+            logger.debug(
+                f"Worker inputs: {worker_inputs}, disaggregated params: {disaggregated_params}"
+            )
+
+            sampling_params = get_sampling_params(request.sampling_params)
+            async for response in self._llm_engine.generate_async(
+                inputs=worker_inputs,
+                sampling_params=sampling_params,
+                disaggregated_params=disaggregated_params,
+                streaming=(self._server_type == ServerType.GEN),
```

Comment:
so always streaming if a gen worker?

curious - why wouldn't we support non streaming in gen case depending on input request?

Reply:
Good point.